artemis/modules/removed_domain_existing_vhost.py

```diff
@@ -0,0 +1,140 @@
+            host, port = address
+            hostname = patch_ip_to
+
+            return _orig_create_connection((hostname, port), *args, **kwargs)  # type: ignore
+
+        connection.create_connection = patched_create_connection
+        return http_requests.get(url)
+
+    def run(self, current_task: Task) -> None:
+        domain = current_task.get_payload("domain")
+
+        if self.check_domain_exists(domain):
+            self.db.save_task_result(task=current_task, status=TaskStatus.OK, status_reason="Domain exists")
+            return
+
+
+        target_ips = self._obtain_past_target_ips(domain)
+        if not target_ips:
+            self.db.save_task_result(
+                task=current_task, status=TaskStatus.OK, status_reason="Unable to obtain past target ips"
+            )
+            return
+
+        prefix = binascii.hexlify(os.urandom(3)).decode("ascii")
+        for ip in target_ips:
+            for proto in ["http", "https"]:
+                try:
+                    response_for_old_domain = self._request_with_patched_ip(proto + "://" + domain, ip)
+                    response_for_other_vhost = self._request_with_patched_ip(proto + "://" + prefix + domain, ip)
+                except requests.exceptions.RequestException:
+                    self.log.exception("Unable to download website content")
+                    continue
+
+                ratio = SequenceMatcher(
+                    None, response_for_old_domain.content, response_for_other_vhost.content
+                ).quick_ratio()
+                self.log.info(f"Similarity between correct and incorrect domain is {ratio} nd status code is {response_for_old_domain.status_code}")
+
+                if (
+                    ratio < Config.Modules.RemovedDomainExistingVhost.REMOVED_DOMAIN_EXISTING_VHOST_SIMILARITY_THRESHOLD
+                    and response_for_old_domain.status_code in range(200, 300)
+                ):
+                    self.db.save_task_result(
+                        task=current_task,
+                        status=TaskStatus.INTERESTING,
+                        status_reason=f"Detected that {ip} hosts nonexistent domain {domain}",
+                        data={
+                            "ip": ip,
+                            "domain": domain,
+                            "response_for_old_domain": response_for_old_domain.content,
+                            "response_for_other_vhost": response_for_other_vhost.content,
+                            "similarity": ratio,
+                        },
+                    )
+                    return
+        self.db.save_task_result(
+            task=current_task,
+            status=TaskStatus.OK,
+            status_reason=f"Didn't detect that any of ips: {target_ips} host {domain}",
+        )
+
+
+if __name__ == "__main__":
+    if (
+        Config.Modules.RemovedDomainExistingVhost.REMOVED_DOMAIN_EXISTING_VHOST_PASSIVEDNS_URL
```

Comment:
I guess, similarity should be checked as well

Reply:
similarity is provided as default, for configs with defaults we don't do such check